chrome/updater/configurator.cc

```diff
@@ -259,20 +308,43 @@ std::vector<uint8_t> Configurator::GetRunActionKeyHash() const {
 }
 
 std::string Configurator::GetAppGuidHelper(const std::string& updater_channel,
-                                           const std::string& version) {
+                                           const std::string& version,
+                                           const int sb_version) {
   if (updater_channel == "ltsnightly" || updater_channel == "ltsnightlyqa") {
     return kOmahaCobaltLTSNightlyAppID;
   }
-  if (version.find(".lts.") != std::string::npos &&
-      version.find(".master.") == std::string::npos) {
-    return kOmahaCobaltAppID;
+  if (version.find(".lts.") == std::string::npos &&
+      version.find(".master.") != std::string::npos) {
+    return kOmahaCobaltTrunkAppID;
+  }
+  std::string channel(updater_channel);
+  if (std::regex_match(updater_channel, std::regex("2[5-9]lts\\d+"))) {
```

Comment:
also add a comment explaining what are you trying to match

Reply:
Added a comment and changed to "\\d+lts\\d+", but I was wondering if it needs to be future proof.

I was going to use "25lts\\d+" since we don't expect Chrobalt to be compatible with SB14-16. These configs will only be used for C25. Given we don't know when we will ship Chrobalt, I used "2[5-9]lts\\d+", but these should really just be used by C25.

Reply:
The static channels should work universally even when setting a device to an older static channel. For example, if a device is running 25lts20 with this code, setting its channel to `24lts10` should work. This means the regex here should match all static channels, and the new config should contain cohorts and rules of all applicable static channels for the specific SB verison.